src/messaging/Message.ts

```diff
@@ -149,16 +190,15 @@ export default class Message implements IMessage {
     }
 
     try {
-      const messageBody = JSON.parse(decoded)
-      return {
-        messageId: encrypted.messageId,
-        receivedAt: encrypted.receivedAt,
+      const messageBody: MessageBody = JSON.parse(decoded)
+      const decrypted: IMessage = {
+        ...encrypted,
         body: messageBody,
-        createdAt: encrypted.createdAt,
```

Comment:
Refactored this to make use of destructuring as `decrypt` was just a clone of encrypted with additional `body` field.